packages/vite-node/src/client.ts

```diff
@@ -289,7 +289,8 @@ export class ViteNodeRunner {
 
     const modulePath = cleanUrl(moduleId)
     // disambiguate the `<UNIT>:/` on windows: see nodejs/node#31710
-    const href = pathToFileURL(modulePath).href
+    const [moduleUrl] = await this.resolveUrl(modulePath)
+    const href = pathToFileURL(moduleUrl).href
```

Comment:
When an aliased modules is loaded the `fsPath` doesn't contain extension at all. @sheremet-va 
 should I make this change to `fsPath` instead so that everywhere it's used would be affected, e.g. stacktraces I think?

With alias, without PR changes:

```js
import { addThreeNumbers } from "src/logic/utils";

{
  __filename: '/x/y/repro/src/logic/utils',
  id: 'src/logic/utils',
  fsPath: 'src/logic/utils'
}
```

Without alias, without PR changes, works OK:

```js
import { addThreeNumbers } from "../../logic/utils";

{
  __filename: '/x/y/repro/src/logic/utils.ts',
  id: '/src/logic/utils.ts',
  fsPath: '/x/y/repro/src/logic/utils.ts'
}
```


With alias, with PR changes:

```js
import { addThreeNumbers } from "src/logic/utils";

{
  __filename: '/x/y/repro/src/logic/utils.ts',
  id: 'src/logic/utils',
  fsPath: 'src/logic/utils'
}
```

Reply:
This should not even be allowed. Vitest should fail if the module is not resolved. I am coming up with a fix for this. Some people rely on tsconfig resolution when in practice we don't support it.

Reply:
I see, the reproduction case is not using `alias` in `vitest.config.ts`. By adding following the coverage works OK:

```ts
    alias: {
      src: fileURLToPath(new URL("./src", import.meta.url)),
    },
```

So I guess in your fix Vitest will fail to load module if only `tsconfig.json` has `alias` but `vitest.config.ts` does not?

Reply:
We can finally stip supporting this weird edge case: https://github.com/vitest-dev/vitest/pull/3307

Reply:
> So I guess in your fix Vitest will fail to load module (or warn?) if only tsconfig.json has alias but vitest.config.ts does not?

People rely on the fact that their modules are relative to the `baseDir` which is never true in Vite. Because of this, we have a lot of issues when the file is not resolved correctly, but it can still fetch module code.